ast/parser_test.go

```diff
@@ -2296,6 +2296,94 @@ func TestParserText(t *testing.T) {
 	}
 }
 
+func TestRuleText(t *testing.T) {
+	input := `
+r1 = x {
+	a == "foo"
+} else = y {
+	b == "bar"
+}
+
+else {
+	c == "baz"
+}
+
+else = {
+	"k1": 1,
+	"k2": 2
+} {
+	true
+}
+`
+
+	rule := MustParseRule(input)
+	assertLocationText(t, strings.TrimSpace(input), rule.Location)
+	assertLocationText(t, "r1 = x", rule.Head.Location)
+	assertLocationText(t, "x", rule.Head.Value.Location)
+
+	curElse := rule.Else
+	if curElse == nil {
+		t.Fatalf("Expected an else block, got nil")
+	}
+	assertLocationText(t, strings.TrimSpace(`
+else = y {
+	b == "bar"
+}
+	`), curElse.Location)
+	assertLocationText(t, "else = y", curElse.Head.Location)
+	assertLocationText(t, "y", curElse.Head.Value.Location)
+
+	curElse = curElse.Else
+	if curElse == nil {
+		t.Fatalf("Expected an else block, got nil")
+	}
+	assertLocationText(t, strings.TrimSpace(`
+else {
+	c == "baz"
+}
+	`), curElse.Location)
+	assertLocationText(t, "else", curElse.Head.Location)
+	if curElse.Head.Value.Location != nil {
+		t.Errorf("Expected a nil location")
+	}
+
+	curElse = curElse.Else
+	if curElse == nil {
+		t.Fatalf("Expected an else block, got nil")
+	}
+	assertLocationText(t, strings.TrimSpace(`
+else = {
+	"k1": 1,
+	"k2": 2
+} {
+	true
+}
+	`), curElse.Location)
+	assertLocationText(t, strings.TrimSpace(`
+else = {
+	"k1": 1,
+	"k2": 2
+}
+	`), curElse.Head.Location)
+	assertLocationText(t, strings.TrimSpace(`
+{
+	"k1": 1,
+	"k2": 2
+}
+	`), curElse.Head.Value.Location)
+}
+
+func assertLocationText(t *testing.T, expected string, actual *Location) {
+	t.Helper()
+	if actual == nil || actual.Text == nil {
```

Comment:
When do we expect nil location text?

Reply:
We check for it explicity up on 2246 for the else head value in 
```
else {
	c == "baz"
}
```

Any calls to this helper are required to specify one (I had it using like pointers or `interface{}` for expected but it got kind of annoying for just that one spot. If we have more we can refactor it to support both types of expected values (nil or string).